golang/internal/grpc/grpc.go

```diff
@@ -446,6 +466,38 @@ func executeSecretList(
 	}
 }
 
+func executeUpdate(ctx context.Context, command *agent.AgentUpdateRequest, updateFunc UpdateFunc) {
+	if updateFunc == nil {
+		log.Warn().Stack().Msg("gRPC self update is not implemented")
+		return
+	}
+
+	if err := updateFunc(ctx, command.Tag); err != nil {
+		log.Error().Stack().Err(err).Msg("Update error")
+
+		resp := &agent.AgentUpdateAborted{
+			Error: err.Error(),
+		}
+
+		_, err := grpcConn.Client.UpdateAborted(ctx, resp)
+		if err != nil {
+			log.Error().Stack().Err(err).Msg("Update error response")
+		}
+	}
+}
+
+func executeClose(ctx context.Context, command *agent.CloseConnection, closeFunc CloseFunc) {
+	log.Debug().Str("reason", agent.CloseReason_name[int32(command.GetReason())]).Msg("gRPC connection remotely closed")
```

Comment:
`CloseReason_name` eh? Camel and snake case mixed?

Reply:
Generated code :)

Reply:
✅